contracts/staking/TokenStaking.sol

```diff
@@ -590,6 +597,54 @@ contract TokenStaking is Initializable, IStaking, Checkpoints {
         emit ToppedUp(stakingProvider, amount);
         increaseStakeCheckpoint(stakingProvider, amount);
         token.safeTransferFrom(msg.sender, address(this), amount);
+
+        if (!stakingProviderStruct.autoIncrease) {
+            return;
+        }
+
+        // increase authorization for all authorized app
+        for (
```

Comment:
The `topUp` function can be called by anyone so, if `autoIncrease` is `true`, anyone can increase staking provider's authorization. Of course, the caller has to transfer an amount, but still, wondering if we aren't opening a potential attack vector here. Shouldn't we do the increase just for callers being provider's authorizers/owners? Curious about your thoughts!

Reply:
Yes, this was intentionally added in https://github.com/threshold-network/solidity-contracts/pull/110.
The rationale was to allow other contracts like Merkle distribution or application contracts to automatically top up stakes.

Reply:
Ok, thanks for clarification!